enterprise/static/enterprise/admin/enterprise_customer_catalog.js

```diff
@@ -0,0 +1,28 @@
+(function($){
+    $(document).ready(function(){
+        var catalogQueryId,
+            // Django admin does auto form filling once the form is loaded.
+            // We want to ignore the first "on change event on enterprise_catalog_query dropdown".
+            // The reason to ignore this is for the case when we might have manually updated the
+            // content_filter field and when laoding the admin page, Django populates the form, fill ups the
+            // dropdown field, which will trigger an "on change event on enterprise_catalog_query dropdown".
+            ignoreUpdate = true,
+            catalogQueryApiUrl = '/enterprise/api/v1/enterprise_catalog_query/',
+            $contentFilterEl = $('#id_content_filter');
+
+        $("[name='enterprise_catalog_query']").on("change", function(event){
+            catalogQueryId = $(event.target).val();
+            if (!ignoreUpdate && catalogQueryId) {
+                // Extract content_filter for the selected catalogQueryId.
+                $.ajax({
+                    url: catalogQueryApiUrl + catalogQueryId,
+                    method: "get"
+                }).success(function (catalogQueryResponse) {
+                    $contentFilterEl.val(JSON.stringify(catalogQueryResponse));
```

Comment:
It would be great if we can indent the content filter? 
https://stackoverflow.com/questions/3515523/javascript-how-to-generate-formatted-easy-to-read-json-straight-from-an-object

Reply:
I am a bit reluctant in indenting this since it may add up spaces which might bring up issue of max_length for content_filter field when saving.